client/src/rr/dnssec/keypair.rs

```diff
@@ -241,11 +242,12 @@ impl KeyPair {
     /// # Return
     ///
     /// the KEY record data
+    #[allow(deprecated)]
     pub fn to_sig0key(&self, algorithm: Algorithm) -> DnsSecResult<KEY> {
         self.to_public_bytes()
             .map(|bytes| {
                      KEY::new(Default::default(),
-                              Default::default(),
+                              KeyUsage::Zone,
```

Comment:
Thanks for your consideration.

I fear I cannot provide any useful inputs, as I'm not really familiar with the DNS RFCs. All I can tell is that `dnssec-keygen` generates keys tagged with `KeyUsage::Zone` on my Debian system. And such keys tend to work in combination with `nsupdate`. I'm not aware of any other use case for a sig0 key (i.e. with `KeyUsage::Entity`), so I don't really feel qualified to cast a vote.

A `to_sig_with_usage()` method in trusted-dns would absolutely satisfy my use case. If you think any other default makes more sense or is better according to the RFCs, please go for that default.